inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp

```diff
@@ -806,28 +815,73 @@ void MKLDNNGraph::Infer(MKLDNNInferRequest* request, int batch) {
         IE_THROW() << "Wrong state. Topology is not ready.";
     }
 
+    if (config.collectPerfCounters) {
+        InferWithPerfCounters(request, batch);
+        return;
+    }
+
     mkldnn::stream stream(eng);
 
     ENABLE_CPU_DEBUG_CAP(NodeDumper nd(config.debugCaps, infer_count));
 
-    for (int i = 0; i < graphNodes.size(); i++) {
-        if (request != nullptr) {
+    for (int i = 0; i < constantGraphNodes.size(); i++) {
+        if (request != nullptr)
             request->ThrowIfCanceled();
-        }
 
-        PERF(graphNodes[i]);
+        if (batch > 0)
+            constantGraphNodes[i]->setDynamicBatchLim(batch);
+
+        ENABLE_CPU_DEBUG_CAP(nd.dumpOutputBlobs(constantGraphNodes[i]));
+    }
+
+    for (int i = 0; i < inconstantGraphNodes.size(); i++) {
+        if (request != nullptr)
+            request->ThrowIfCanceled();
 
         if (batch > 0)
-            graphNodes[i]->setDynamicBatchLim(batch);
+            inconstantGraphNodes[i]->setDynamicBatchLim(batch);
 
-        ENABLE_CPU_DEBUG_CAP(nd.dumpInputBlobs(graphNodes[i]));
+        ENABLE_CPU_DEBUG_CAP(nd.dumpInputBlobs(inconstantGraphNodes[i]));
 
-        if (!graphNodes[i]->isConstant()) {
-            OV_ITT_SCOPED_TASK(itt::domains::MKLDNNPlugin, graphNodes[i]->profiling.execute);
-            graphNodes[i]->execute(stream);
-        }
+        OV_ITT_SCOPED_TASK(itt::domains::MKLDNNPlugin, inconstantGraphNodes[i]->profiling.execute);
+        inconstantGraphNodes[i]->execute(stream);
+
+        ENABLE_CPU_DEBUG_CAP(nd.dumpOutputBlobs(inconstantGraphNodes[i]));
+    }
+
+    if (infer_count != -1) infer_count++;
+}
+
+void MKLDNNGraph::InferWithPerfCounters(MKLDNNInferRequest* request, int batch) {
```

Comment:
Looks like there is a lot of code duplication with `Infer()` method. Not good.

Reply:
I wanted to avoid frequent conditions in cycle
But I have checked It seems like there is no overhead. So I added If statement
Thank you!